jerry-ext/include/jerryscript-ext/section.impl.h

```diff
@@ -0,0 +1,84 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_SECTION_IMPL_H
+#define JERRYX_SECTION_IMPL_H
+
+/**
+ * Define the name of a section
+ *
+ * @param name the name of the section without quotes
+ */
+#ifdef __MACH__
+#define JERRYX_SECTION_NAME(name) "__DATA," #name
```

Comment:
@zherczeg Also, `-U__GNUC__` != extensions disabled, because if `__attribute__((something))` appears unconditionally, gcc will still accept that. So, it remains our responsibility to provide for the case where `__GNUC__` is not defined and then to test for that case using travis.

I actually wonder if Ubuntu has a C compiler that does not provide GNU extensions. Then we could do a travis job where we use that as the compiler instead of the default one.